New article and content for `Orleans.BroadcastChannel` NuGet package (#33044)
dotnet/docs: docs/orleans/streaming/streams-quick-start.md

---
title: Orleans streaming quickstart
description: Learn from the streaming quickstart in .NET Orleans.
ms.date: 01/05/2023
zone_pivot_groups: orleans-version
---

# Orleans streaming quickstart

This guide will show you a quick way to set up and use Orleans Streams. To learn more about the details of the streaming features, read other parts of this documentation.

## Required configurations

<!-- markdownlint-disable MD044 -->
:::zone target="docs" pivot="orleans-7-0"

> <!-- markdownlint-enable MD044 -->

Reviewer: This is in other places in addition to this one, but MD044 is [Proper names should have the correct capitalization](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md044---proper-names-should-have-the-correct-capitalization). The rules for tabs are MD024/25. However, the config in the repo has them disabled, so I don't think you even need this.

Author: It's complaining that the zone pivot identifier, `orleans-` isn't capitalized as it's marked as a proper name.

> In this guide, we'll use a memory-based stream that uses grain messaging to send stream data to subscribers. We will use the in-memory storage provider to store lists of subscriptions, so it is not a wise choice for real production applications.

Reviewer: I don't really understand why this isn't a wise choice. Is it because it's not persisted in some backend and could be lost? Are you suggesting that this is good for testing but not production? I think the guidance here could be a bit clearer. You may even suggest an alternative that is better with the reason as to why you're using the memory provider.

Author: I'll clarify that the memory stream is intended for dem, local development, and test purposes only. Since if the process goes down, none of the streaming content is persisted.

> In this guide, we'll use a memory-based stream that uses grain messaging to send stream data to subscribers. We will use the in-memory storage provider to store lists of subscriptions, so it is not a wise choice for real production applications.

Author: ```suggestion
In this guide, you'll use a memory-based stream that uses grain messaging to send stream data to subscribers. You will use the in-memory storage provider to store lists of subscriptions. Using memory-based mechanisms for streaming and storage is only intended for local development and testing, and isn't intended for production environments.
```

> On the silo, where `silo` is an <xref:Orleans.Hosting.ISiloBuilder> call <xref:Orleans.Hosting.SiloBuilderExtensions.AddMemoryStreams%2A>:

Reviewer: ```suggestion
On the silo, where `silo` is an <xref:Orleans.Hosting.ISiloBuilder>, call <xref:Orleans.Hosting.SiloBuilderExtensions.AddMemoryStreams%2A>:
```

```csharp
silo.AddMemoryStreams("StreamProvider")
    .AddMemoryGrainStorage("PubSubStore");
```

> On the cluster client, where `client` is an <xref:Orleans.Hosting.IClientBuilder> call <xref:Orleans.Hosting.ClientBuilderStreamingExtensions.AddMemoryStreams%2A>.

Reviewer: ```suggestion
On the cluster client, where `client` is an <xref:Orleans.Hosting.IClientBuilder>, call <xref:Orleans.Hosting.ClientBuilderStreamingExtensions.AddMemoryStreams%2A>.
```

```csharp
client.AddMemoryStreams("StreamProvider");

:::zone-end

<!-- markdownlint-disable MD044 -->
:::zone target="docs" pivot="orleans-3-x"
<!-- markdownlint-enable MD044 -->

In this guide, we'll use a simple message-based stream that uses grain messaging to send stream data to subscribers. We will use the in-memory storage provider to store lists of subscriptions, so it is not a wise choice for real production applications.

> On the silo, where `hostBuilder` is an `ISiloHostBuilder` call <xref:Orleans.Hosting.StreamHostingExtensions.AddSimpleMessageStreamProvider%2A>:

Reviewer: ```suggestion
On the silo, where `hostBuilder` is an `ISiloHostBuilder`, call <xref:Orleans.Hosting.StreamHostingExtensions.AddSimpleMessageStreamProvider%2A>:
```

```csharp
hostBuilder.AddSimpleMessageStreamProvider("SMSProvider")
           .AddMemoryGrainStorage("PubSubStore");
```

> On the cluster client, where `clientBuilder` is an `IClientBuilder` call <xref:Orleans.Hosting.ClientStreamExtensions.AddSimpleMessageStreamProvider%2A>.

Reviewer: ```suggestion
On the cluster client, where `clientBuilder` is an `IClientBuilder`, call <xref:Orleans.Hosting.ClientStreamExtensions.AddSimpleMessageStreamProvider%2A>.
```
